File: class_Member/2.rb
#루비에서 메소드가 클래스의 소속이라면 실행할때 클래스소속이라고 표시해야합니다.
#클래스 멤버는 인스턴스에서 사용할수없을때, 인스턴스에서도 클래스에서도 사용할수없다.
#클래스는 인스턴스소속에 속한다, 인스턴스소속은 클래스에 소속된다.

class Cs
  def Cs.class_method()
    p "Class method"
  end
    def instance_method()
      p "Instance_method"
  end
end
i = Cs.new()
Cs.class_method()
i.instance_method() #i = cs를 인스턴스화 한것을 i이다.
# Cs.instance_method()
# i.class_method()

File: class_Member/2.py
#클래스멤버에 종류에는 클래스 메소드와 스태틱 메소드가 있다.
#인스턴스에 소속된것은 인스턴스멤버다.
#cs = 클래스
#i = 인스턴스
class Cs:
    @staticmethod
    def static_method():
        print("Static method")
    @classmethod
    def class_method(cls):          #첫번째 매개변수(cls)가 있어야합니다.
        print("Class method")
    def instance_method(self):      #첫번째 인자값(self)을 준다.
        print("Instance_method")
i = Cs()
Cs.static_method()                  #클래스 멤버는 스태틱메소드의 멤버이다.
Cs.class_method()                   #클래스 멤버는 클래스메소드의 멤버이다.
i.instance_method()                 #인스턴스 멤버는 인스턴스의 멤버이다.
